src/dmd/traits.d

```diff
@@ -1453,28 +1453,16 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
 
         if (l1 && l2)
         {
-            const(char)* getSerialization(FuncLiteralDeclaration fld)
+            import dmd.lambdacomp : getSerialization;
+            if (auto ser1 = getSerialization(l1, sc))
             {
-                import dmd.lambdacomp : SerializeVisitor;
-                auto serVisitor = new SerializeVisitor(sc);
-                fld.accept(serVisitor);
-                if (serVisitor.buf.offset == 0)
-                    return "uncomparable";
-                else
-                    return serVisitor.buf.extractString();
-            }
-
-            if (!l1.serialization)
-                l1.serialization = getSerialization(l1);
-            if (!l2.serialization)
-                l2.serialization = getSerialization(l2);
-
-            //printf("serialization1: %s\n", l1.serialization);
-            //printf("serialization2: %s\n", l2.serialization);
-            if (strcmp(l1.serialization, l2.serialization) == 0 &&
-                strcmp(l1.serialization, "uncomparable") != 0)
-            {
-                return True();
+                //printf("l1 serialization: %s\n", ser1);
+                if (auto ser2 = getSerialization(l2, sc))
+                {
+                    //printf("l2 serielization: %s\n", ser2);
+                    if (ser1 == ser2)
+                        return True();
+                }
```

Comment:
Note that the allocated memory returned by `getSerialization` is no longer needed here. This suggests a further encapsulation of the lines 1456 to here, create a new function `isSameFuncLiteral()` in `lambdacomp.d` that returns `true` if they are the same. In that function, put the `OutBuffer` declarations, so the memory is recovered at the function end.